src/features/Classes/data/thunks.js

```diff
@@ -7,13 +7,24 @@ import {
 } from 'features/Classes/data/slice';
 import { getClassesByInstitution } from 'features/Common/data/api';
 
-function fetchClassesData(id, currentPage, courseName = '') {
+function fetchClassesData(id, currentPage, courseName = '', urlParamsFilters = '', instructor = '', limit = true) {
   return async (dispatch) => {
     dispatch(fetchClassesDataRequest);
 
     try {
-      const response = camelCaseObject(await getClassesByInstitution(id, courseName, true, '', currentPage));
-      dispatch(fetchClassesDataSuccess(response.data));
+      const response = camelCaseObject(
+        await getClassesByInstitution(id, courseName, limit, instructor, currentPage, urlParamsFilters),
+      );
+      if (limit) {
```

Comment:
Why you are making this validation? For the case of the table the limit should always be true to get the pagination, and if we don't need the pagination we should store that data in other part of the store